### PR TITLE
AT-84-file-upload-history

### PR DESCRIPTION
**Pull request by b7rch4**
Co-authored-by: richardtully richardtully12@gmail.com

Jira: https://dfacademy.atlassian.net/jira/software/projects/AT/boards/4?selectedIssue=AT-84

Problem:  We need to display the number of file uploads that have occurred over the last 7 days. This should be grouped by type.

Solution: We have updated the response when the user visits /dashboard so that it will contain the required data.
We query the database to find all files uploaded in the past 7 days, then transform the data into the format required by the frontend.

### CODE REVIEW

File: src/dashboard/countUploadsPerDay.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+const countUploadsPerDay = (dates) => {
```
Comment (dearshrewdwit) [inline src/dashboard/countUploadsPerDay.js:1]: This is quite a fixed implementation. What about every 10 days, or over the whole month?

Comment (dearshrewdwit) [reply src/dashboard/countUploadsPerDay.js:1]: How would you decouple this function from knowing about how many days?

File: src/dashboard/fileUploads.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  date.setDate(date.getDate()-6);
+  date.setUTCHours(0);
+
+  const uploadQuery = await UploadHistory.findAll({
+    raw: true,
+    attributes: ["uploadType", "createdAt"],
+    where: {
+      status: "success",
+      createdAt: {
+        [Op.gt]: date
```
Comment (dearshrewdwit) [inline src/dashboard/fileUploads.js:16]: Nice

File: src/dashboard/countUploadsPerDay.js (hunk 1 above)
Comment (richardtully) [reply src/dashboard/countUploadsPerDay.js:1]: Yes, I think in the back of our minds we were expecting this comment!

File: src/dashboard/countUploadsPerDay.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+const countUploadsPerDay = (dates, days) => {
```
Comment (dearshrewdwit) [inline src/dashboard/countUploadsPerDay.js:1]: Ok - better for decoupling knowing about days.

Currently this implementation is still quite opaque - I can't read this and understand it. I have to console.log my way through this to understand what is happening. Maybe my morning coffee hasn't kicked in.

As an experiment - give these 30 lines of code to a non-team member and ask them to walk you through it. Code should be able to be read an understand by other people - it's why we write clean code. If they struggle - that's some information for you to think about how maintainable this is.

Comment (richardtully) [reply src/dashboard/countUploadsPerDay.js:1]: Ok, thanks, we'll try to make it more readable

**Review by dearshrewdwit — APPROVED** [review]
I'll approve for now to unblock this. Ordinarily, we would have workshopped this to clean it up.

File: src/dashboard/countUploadsPerDay.js (hunk 3 above)
Comment (richardtully) [reply src/dashboard/countUploadsPerDay.js]: I think we've made it more simple but it's hard to tell from the perspective of someone who made it. We did get another team member to check it and they said it seemed ok. We're interested to hear what you think of it.

File: src/dashboard/countUploadsPerDay.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+
```
Comment (dearshrewdwit) [inline src/dashboard/countUploadsPerDay.js:1]: Better.

**Review by dearshrewdwit — APPROVED** [review]

**Review by FractalFraction — APPROVED** [review]

**Pull request merged**